design-patterns/creational-patterns/src/objectPool/ReusablePool.java

```diff
@@ -0,0 +1,37 @@
+package objectPool;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReusablePool {
+	private List<Reusable> reusables;
+	private static ReusablePool instance;
+
+	private ReusablePool(int size){
+		reusables = new ArrayList<Reusable>(size);
+		
+		for(int i=0;i<size;i++)
+			reusables.add(new Reusable());
```

Comment:
Try to avoid logic in constructors. I would suggest that to provide a list or objects to the constructor that the pool should include.

Reply:
is it ok to call a method from constructor, (outside setters)?

Reply:
My comment seems it is outdated. No need to be fixed for now.